packages/apps/auth0-actions/src/redirect_to_full_registration.ts

```diff
@@ -0,0 +1,65 @@
+import { Auth0User } from '@weco/auth0-client';
+import { Event, API } from './types/post-login';
+
+const REGISTRATION_FORM_URL = 'http://localhost:3000/account/registration';
+const SUCCESS_URL = 'http://localhost:3000/account';
+
+export const onExecutePostLogin = async (
+  event: Event<Auth0User>,
+  api: API<Auth0User>
+) => {
+  // If the user has accepted the terms already, we don't need to do anything else so bail out here
+  if (event.user.app_metadata?.terms_and_conditions_accepted) return;
+
+  // We send the user straight to the full registration form after they first register email and password
+  // Full registration form has three fields aka formData: firstname, lastname and terms_and_conditions_accepted
+  // In Identity App (Weco) we post the formData to updateUserAfterRegistration endpoint on Identity API
+  // updateUserAfterRegistration updates auth0 user with formData firstname and lastname
+  // updateUserAfterRegistration calls sierra http client and creates a patron with firstname and lastname
+  // updateUserAfterRegistration encodes formData and sends back to this auth0 action
+  // auth0 action we update terms_and_conditions_accepted boolean
+  // auth0 action will then redirect to success page
+
+  const sessionToken = api.redirect.encodeToken({
+    // this token must match the one used in Identity API
+    secret: event.secrets.AUTH0_ACTION_SECRET,
+    payload: {
+      iss: `https://${event.request.hostname}/`,
```

Comment:
it should always come from Auth0 and should come with a state value. The state value being unique, as well as secret being shared between action and Identity weco, should be enough to prevent injection by bad actors https://auth0.com/docs/secure/attack-protection/state-parameters